### PR TITLE
Add local spawning of workers. Fixes #64

### LINKED ISSUE #64: Add an `aimmo-game` implementation that spawns workers locally
i.e. replace the current method of speculatively starting 4 workers.
Some work here: https://github.com/jishi9/aimmo/commit/9c8629135654b1940e5a925bb88dc02e79c1cca7

### PR DESCRIPTION
**Pull request by mikebryant**

### CODE REVIEW

File: aimmo-game/simulation/worker_manager.py [hunk 1]
```diff
@@ -30,22 +44,64 @@ def run(self):
...
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.workers = {}
+        self.next_port = 1989
+        super(LocalWorkerManager, self).__init__(*args, **kwargs)
+
+    def create_worker(self, player_id):
+        if player_id in self.workers:
+            raise NotImplemented
```
Comment (jishi9) [inline aimmo-game/simulation/worker_manager.py]: should really be an assertion

File: aimmo-game/simulation/worker_manager.py [hunk 2]
```diff
@@ -30,22 +44,64 @@ def run(self):
...
+                self.host,
+                str(self.next_port),
+            ],
+        )
+        worker_url = 'http://%s:%d' % (
+            self.host,
+            self.next_port,
+        )
+        LOGGER.info("Worker started for %s, listening at %s", player_id, worker_url)
+        time.sleep(5)
```
Comment (jishi9) [inline aimmo-game/simulation/worker_manager.py]: Is this to give time to the player to start?

Comment (mikebryant) [inline aimmo-game/simulation/worker_manager.py]: Aye. This should be some sort of retry logic, and initialisation should probably be split off from worker creation. I'll raise another issue for this

File: aimmo-game/simulation/worker_manager.py [hunk 3]
```diff
@@ -30,22 +44,63 @@ def run(self):
...
+                for removed_user in removed_user_ids:
+                    self.game_state.remove_avatar(removed_user)
+                    self.remove_worker(removed_user)
+                    del self.user_codes[removed_user]
 
             time.sleep(10)
+
+class LocalWorkerManager(WorkerManager):
+    '''Relies on them already being created already.'''
+    host = '127.0.13.4'
```
Comment (jishi9) [inline aimmo-game/simulation/worker_manager.py]: doesn't work on mac :(

**Comment by jishi9** [comment]
Looks good, runs locally for me.

**Pull request merged**